web/main/views.py

```diff
@@ -115,8 +115,8 @@
 ### helpers ###
 
 
-def login_required_response(request):
-    if request.user.is_authenticated:
+def login_required_response(request: HttpRequest, always_raise=False):
+    if request.user.is_authenticated or always_raise:
```

Comment:
Without this, attempts to hit a professor-only resource by an anonymous user would prompt to log in rather than return Forbidden. For consistency with the logged-in but not authorized experience, I went with forcing a 403 rather than a login prompt here.